packages/backend/ubuntu-pro.go

```diff
@@ -0,0 +1,20 @@
+package main
+import (
+    "os/exec"
+    "context"
+    "google.golang.org/grpc/codes"
+    "google.golang.org/grpc/status"
+    pb "github.com/canonical/desktop-security-center/packages/proto"
+    "google.golang.org/protobuf/types/known/emptypb"
+    "github.com/tidwall/gjson"
+)
+
+func (s *ProServer) IsMachineProAttached(ctx context.Context, _ *emptypb.Empty) (*pb.Boolean, error) {
+    cmd := exec.Command("pro", "api", "u.pro.status.is_attached.v1")
+    out, err := cmd.Output()
+    if err != nil {
+        return nil, status.Errorf(codes.NotFound, "couldn't query Pro status")
+    }
+    val := gjson.Get(string(out), "data.attributes.is_attached")
+    return &pb.Boolean{ Ret: val.Bool() }, nil
```

Comment:
This is returning gibberish. In my [simple test client](https://people.canonical.com/~npt/client.go), the call returns

```
&{{{} [] [] %!t(*impl.MessageInfo=&{0x871500 0xc0000e39d0 <nil> [] {0 0} 1 {map[1:0xc000296280] map[] map[] [<nil> 0xc000296280] [0xc000296280] 0x601520 0x601480 0x60cf40 {<nil>}} {{{} 3 0x612100 0x612020 0x612180 0x612320 0x6122a0} [0xc000298240] [<nil> 0xc000298240] map[1:0xc000298240] 8 16 false 18446744073709551615 false false 0}})} %!t(int32=0) [] false}
```

How is this a boolean?!

Reply:
I don't think the response will be just a bool, it will probably have some wrapping, so trying to print it with `%t` is what I think causing the issue:

You could try printing the full response struct and its field to get a better read of what is happening:

```
    u, err := p.IsMachineProAttached(context, new(emptypb.Empty))
	if err != nil {
		fmt.Println(err)
	} else {
        fmt.Printf(":%+v:%t\n", u, true)
    }
```

Alternatively,  you could use [gRPCurl](https://github.com/fullstorydev/grpcurl) to test, which will do the formatting for you, and you don't need to extend it each time you add a new endpoint you want to test:

```
grpcurl -plaintext -unix -proto locale.proto -d '{"locale": ""}' /home/matt/test.sock locale.LocaleService/SetLocale
```
In the above example:
`locale.proto` = path to the protofile that defines the endpoint you want to test
`'{"locale":""}'` = your request 
`/home/matt/test.sock` = path to your unix socket
`locale.LocaleService/SetLocale` = `proto package name` . `service name` / `rpc name` (all as defined in your protofile)

Reply:
Thank you for your help, Matt!

I hadn't realized that the result would be in some wrapping.

If I try to print it as suggested, it prints:

```
::true
```

I'll try the tool you suggested and report back.

Reply:
Great! Glad that helped :smile:

Reply:
For the record, `grpcurl` agrees with client.go: it is returning `{}`.

Reply:
OK, if I read https://stackoverflow.com/questions/75352422/grpc-not-returning-boolean-if-value-is-false correctly this is by design. Unbelievable!